Week 1 Section 1

GaxysGalactic opened the pull request:
My proposed solutions to Week 2 Section 1 exercises, now in the new repository!


#E1
![E1](https://user-images.githubusercontent.com/44332093/178816212-6b9aee20-17f7-4df7-ae4b-887b46e611b1.PNG)


#E2
![E2](https://user-images.githubusercontent.com/44332093/178816246-cd48f251-1407-4bc5-9853-918533e00c36.PNG)


#E3
![E3](https://user-images.githubusercontent.com/44332093/178816267-7d3674a0-b186-4e65-959c-0e97c62c44e3.PNG)


#E4
![E4](https://user-images.githubusercontent.com/44332093/178816292-7c3fee6b-aaab-4c05-b4be-50aae8c560ad.PNG)

GaxysGalactic committed:
Week 1 Section 1

My proposed solutions to Week 1 Section 1 exercises, now in the new repository!

GaxysGalactic committed:
Comments and Moves

Moved files into their intended folder. Also added comments which explain the exercise.

JDAlzate commented on Month1/Week2/Section1/Exercise1/Main.cpp:
```diff
@@ -0,0 +1,57 @@
+
+void testOne()
+{
+	FirstClass fc1;
```
Mind your names. Remember to use explicit names. What would you understand if you saw the name fc1 alone?

JDAlzate commented on Month1/Week2/Section1/Exercise1/Main.cpp:
```diff
@@ -0,0 +1,57 @@
+{
+	for (int i = 0; i < 5; i++)
+	{
+		FirstClass f6;
```
Same (and especially) here. What is f6 supposed to mean?

JDAlzate commented on Month1/Week2/Section1/Exercise3/Main.cpp:
```diff
@@ -0,0 +1,154 @@
+
+	float getPerimeter() const
+	{
+		return 4 * sideLength;
```
Remember we use .f for floats. If you want to output a float, why use an integer instead of a float for this multiplication? Same as the rest of integers in this file that could be floats.

JDAlzate reviewed (changes requested):
Just a couple comments :)

GaxysGalactic committed:
Name changes and Floats

- [FIX] Changed ints in Exercise3 to floats (by adding f at the end)
- [FIX] Changed the names of the instances in Exercise1 to be more readable, though they still remain pretty generic.

JDAlzate commented on Month1/Week2/Section1/Exercise1/Main.cpp:
```diff
@@ -28,19 +28,19 @@ class FirstClass
 	for (int i = 0; i < 5; i++)
 	{
-		FirstClass f6;
+		FirstClass entitySix;
```
All the first ones make sense since they are in the same scope. However, keep in mind that functions have their own scope. There isn't really a reason to call this entitySix for two reasons:
1. You are creating this 5 times. Do you have 5 `entity6`?
2. There are no other entities in this function. There is no concept of entities 1-5 in the current scope. A project can have literal millions of lines of code, you wouldn't want to check if somewhere in the code there is an `entity1` to see if you should name your entity `entity2`. In your context, this is the only entity. What do you think?

GaxysGalactic replied:
Got it! Will make the changes soon!

JDAlzate reviewed (changes requested):
Just one extra comment :)

GaxysGalactic committed:
Change variable name

- [FIX] Changed variable name on Exercise 1 to be more mindful of scope.

JDAlzate reviewed (approved):
Cool! Merge away!
